src/lib/ast.rs

```diff
@@ -0,0 +1,623 @@
+// Copyright (c) 2017 King's College London
```

Comment:
OK, well please add the year to the copyright, unify the messages and remember to bump the copyright year if you see on out of date ;)